### PR TITLE
Add embeddable summary to CoC performance report

### PR DESCRIPTION
**Pull request by dtgreiner**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Add the summary as a standalone embeddable report for the CoC Performance Measurement report.

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [X] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [X] I have updated the documentation (or not applicable)
- [X] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 1]
```diff
@@ -1373,6 +1373,10 @@ def spm_fields
     end
 
     # Publishing
+    def publish_summary?
+      true
+    end
+
     def publish_files
       [
         {
```
Comment (eanders) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb:1410]:
```suggestion
        {
           name: 'summary.html',
           content: -> { summary_as_html },
           type: 'text/html',
         },
        {
```

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 2]
```diff
@@ -1428,6 +1432,61 @@ def publish_files
...
+          content: -> { Rails.application.assets['icons.woff'].to_s },
+          type: 'text/css',
+        },
+        {
+          name: 'icons.woff2',
+          content: -> { Rails.application.assets['icons.woff'].to_s },
+          type: 'text/css',
+        },
+      ]
+    end
```
Comment (eanders) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb]: Don't need this.

File: app/models/concerns/warehouse_reports/s3_toolset.rb [hunk 3]
```diff
@@ -103,6 +103,25 @@ def ready_public_s3_bucket!
...
+          content_disposition: 'inline',
+          content_type: file[:type],
+        )
+        if resp.etag
+          Rails.logger.info "Successfully uploaded summary report file to s3 (#{key})"
+        else
+          Rails.logger.info "Unable to upload summary report file (#{key}})"
+        end
+      end
+    end
```
Comment (eanders) [inline app/models/concerns/warehouse_reports/s3_toolset.rb]: Don't need this

File: app/models/concerns/warehouse_reports/s3_toolset.rb [hunk 4]
```diff
@@ -142,6 +161,21 @@ def ready_public_s3_bucket!
...
+          bucket: bucket,
+          key: key,
+        )
+        if resp.delete_marker
+          Rails.logger.info "Successfully removed summary report file from s3 (#{key})"
+        else
+          Rails.logger.info "Unable to remove the summary report file (#{key})"
+        end
+      end
+    end
```
Comment (eanders) [inline app/models/concerns/warehouse_reports/s3_toolset.rb]: Don't need this

File: app/models/concerns/warehouse_reports/publish.rb [hunk 5]
```diff
@@ -23,6 +23,22 @@ def publish_files
...
+    def publish_summary_files
+      [
+        {
+          name: 'index.html',
+          content: -> { summary_as_html },
+          type: 'text/html',
+        },
+      ]
+    end
+
```
Comment (eanders) [inline app/models/concerns/warehouse_reports/publish.rb]: Don't need these

File: app/models/concerns/warehouse_reports/publish.rb [hunk 6]
```diff
@@ -52,6 +68,21 @@ def generate_publish_url
...
+      published_report.published_url.split('/').insert(-2, 'summary').join('/')
+    end
+
+    def publish_summary_embed_code
+      return unless publish_summary?
+      return unless published_report.present?
+
+      published_report.embed_code.gsub(published_report.published_url, publish_summary_url)
+    end
+
```
Comment (eanders) [inline app/models/concerns/warehouse_reports/publish.rb:85]: Don't need this

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 7]
```diff
@@ -1373,6 +1373,10 @@ def spm_fields
     end
 
     # Publishing
+    def publish_summary?
+      true
+    end
```
Comment (eanders) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb:1378]: Don't need this

File: drivers/performance_measurement/app/models/performance_measurement/report.rb [hunk 8]
```diff
@@ -1428,6 +1432,61 @@ def publish_files
       ]
     end
 
+    def publish_summary_files
```
Comment (eanders) [inline drivers/performance_measurement/app/models/performance_measurement/report.rb]: Don't need this method.

File: app/models/concerns/warehouse_reports/publish.rb [hunk 9]
```diff
@@ -123,10 +154,24 @@ def as_html
...
+    def summary_as_html
+      return controller_class.render(view_summary_template, layout: raw_layout, assigns: { report: self }) unless view_template.is_a?(Array)
+
+      view_template.map do |template|
+        string = html_section_start(template)
+        string << controller_class.render(template, layout: raw_layout, assigns: { report: self })
+        string << html_section_end(template)
+      end.join
+    end
+
```
Comment (eanders) [inline app/models/concerns/warehouse_reports/publish.rb]: Move this into `performance_measturement/report.rb`

File: app/models/concerns/warehouse_reports/publish.rb [hunk 10]
```diff
@@ -123,10 +154,24 @@ def as_html
...
     def view_template
       :raw
     end
 
+    def view_summary_template
+      :raw_summary
+    end
```
Comment (eanders) [inline app/models/concerns/warehouse_reports/publish.rb]: Move this into `performance_measturement/report.rb`

File: app/models/concerns/warehouse_reports/publish.rb [hunk 11]
```diff
@@ -52,6 +68,21 @@ def generate_publish_url
...
+      # Reference the summary folder within the S3 directory for this published report
+      published_report.published_url.split('/').insert(-2, 'summary').join('/')
+    end
+
+    def publish_summary_embed_code
+      return unless publish_summary?
+      return unless published_report.present?
+
+      published_report.embed_code.gsub(published_report.published_url, publish_summary_url)
+    end
```
Comment (eanders) [inline app/models/concerns/warehouse_reports/publish.rb]: move these into the report.rb file

File: app/models/concerns/warehouse_reports/publish.rb [hunk 12]
```diff
@@ -52,6 +68,21 @@ def generate_publish_url
       "#{publish_url}/index.html"
     end
 
+    def publish_summary_url
+      return unless publish_summary?
+      return unless published_report.present?
+
+      # Reference the summary folder within the S3 directory for this published report
+      published_report.published_url.split('/').insert(-2, 'summary').join('/')
```
Comment (eanders) [inline app/models/concerns/warehouse_reports/publish.rb]: Will need to get updated to replace `index.html` with `summary.html`

**Comment by dtgreiner** [comment]
@eanders The changes we went over have been pushed to the PR

**Review by eanders — APPROVED** [review]
Amazing, thank you @dtgreiner !

**Pull request merged**